src/Microsoft.AspNetCore.Mvc.RazorPages/RedirectToPageResult.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.RazorPages.Internal;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages
+{
+    /// <summary>
+    /// An <see cref="ActionResult"/> that returns a Found (302)
+    /// or Moved Permanently (301) response with a Location header.
+    /// Targets a registered route.
+    /// </summary>
+    public class RedirectToPageResult : ActionResult, IKeepTempDataResult
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RedirectToPageResult"/> with the values
+        /// provided.
+        /// </summary>
+        /// <param name="pageName">The page to redirect to.</param>
+        public RedirectToPageResult(string pageName)
```

Comment:
Should this be `page` or `pageName`? it would be nice to be consistent across usages.